h2o-algos/src/main/java/hex/glm/DispersionUtils.java

```diff
@@ -188,14 +192,36 @@ private static double goldenRatioDispersionSearch(GLMModel.GLMParameters parms,
      */
     public static double estimateTweedieDispersionOnly(GLMModel.GLMParameters parms, GLMModel model, Job job,
                                                               double[] beta, DataInfo dinfo) {
-        long currTime = System.currentTimeMillis();
-        long modelBuiltTime = currTime - model._output._start_time;
-        long timeLeft = parms._max_runtime_secs > 0 ? (long) (parms._max_runtime_secs * 1000 - modelBuiltTime)
-                : Long.MAX_VALUE;
-        TweedieMLDispersionOnly tDispersion = new TweedieMLDispersionOnly(parms.train(), parms, model, beta, dinfo);
+        if (parms._tweedie_variance_power >= 2 && 
+            dinfo._adaptedFrame.vec(parms._response_column).min() <= 0) {
+            Log.warn("Response contains zeros or negative values but "+
+                    "Tweedie variance power does not support zeros. "+
+                    "Instances with response <= 0 will be skipped.");
+            model.addWarning("Response contains zeros or negative values but "+
+                    "Tweedie variance power does not support zeros. "+
+                    "Instances with response <= 0 will be skipped.");
+        }
         DispersionTask.GenPrediction gPred = new DispersionTask.GenPrediction(beta, model, dinfo).doAll(
                 1, Vec.T_NUM, dinfo._adaptedFrame);
         Vec mu = Scope.track(gPred.outputFrame(Key.make(), new String[]{"prediction"}, null)).vec(0);
+        List<Double> logLikelihoodSanityChecks = new ArrayList<>();
+        List<Double> dispersionsSanityChecks = new ArrayList<>();
+        logLikelihoodSanityChecks.add(getTweedieLogLikelihood(parms, dinfo, parms._init_dispersion_parameter, mu));
+        dispersionsSanityChecks.add(parms._init_dispersion_parameter);
+        final double dispersion = goldenRatioDispersionSearch(parms, dinfo, mu, logLikelihoodSanityChecks, dispersionsSanityChecks, job);
+        Log.info("Tweedie dispersion estimate = "+dispersion);
+        return dispersion;
+        
+        /*
+        // FIXME: The Newton's method seems not to be reproducible on jenkins (runit_GLM_tweedie_ml_dispersion_estimation_only.R)
```

Comment:
After consultation with @wendycwong we decided to disable the Newton's method due to some non-reproducible behavior.

---
More details here:

I found out that the tweedie test [runit_GLM_tweedie_ml_dispersion_estimation_only.R](https://github.com/h2oai/h2o-3/blob/35f65b80598eb0e0c654fa71571e5561d54a45ce/h2o-r/tests/testdir_algos/glm/runit_GLM_tweedie_ml_dispersion_estimation_only.R) is now often failing on jenkins (that’s the test that forced me to add the tweedie package to the test images thus creating this PR).

I fixed seeds for generating data and model training and I get the same models (coefficients) locally and on jenkins. I was not able to reproduce it locally or on docker on mr-dl18 with 4 instances and the same test image that should run on jenkins.

For the rest of this message I will describe the first failing test case - `test_p1.6_phi100_no_offset`.

### Coefficients:
**Jenkins:**

Rfit:
```
(Intercept)          xt 
  0.9266287   1.0217314 
```

H2Ofit:
```
Coefficients: glm coefficients
      names coefficients std_error    z_value  p_value
1 Intercept     0.926628  0.080632  11.492105 0.000000
2        xt     1.021732  0.009556 106.920895 0.000000
  standardized_coefficients
1                  6.546151
2                  2.934847
```

**Local:**

Rfit:
```
(Intercept)          xt 
  0.9266287   1.0217314 
```

H2Ofit:
```
Coefficients: glm coefficients
      names coefficients std_error    z_value  p_value
1 Intercept     0.926628  0.083351  11.117132 0.000000
2        xt     1.021732  0.009878 103.432206 0.000000
  standardized_coefficients
1                  6.546151
2                  2.934847
```

So I think the seed worked and I got the same data and the same model so I think the issue is in the dispersion estimation and it appears to be in the optimization part that uses Newton's method - I added a `Log.info` to the Newton’s method to log Tweedie dispersion ML estimation [iter=x, phi=y]  and the behavior seems to diverge there - my test fails when the Newton's method code converges relatively close to the true value (phi should be 100 and it converges to 95 on jenkins and locally to 4224, so locally I switch to gradient-free optimization and converge to 101 (which is almost equal to R’s value)).

The difference in negative log-likelihood is small but it’s more visible in the dispersion estimation.
```
H2O negative log-likelihood: 43141.75
R negative log-likelihood: 43131.49
H2O is better: FALSE
H2O and R are roughly similar: FALSE
H2O Dispersion Estimation: 95.16375
R Dispersion Estimation: 101.6916
H2O is as close as R or closer to the true dispersion: FALSE
```

**Jenkins:**
```
01-19 15:08:49.674 127.0.0.1:40006       238       FJ-1-13  INFO water.default: GLM[dest=TweedieDispersionMLE_p1.6_phi100_offset0, iter=5 lmb=.0E0 alpha=.5E0 obj=105.36 imp=.11E-11 bdf=.49E-5] computed in 5+0=5ms, step = 1
01-19 15:08:49.674 127.0.0.1:40006       238       FJ-1-13  INFO water.default: estimating dispersion parameter using method: ml
01-19 15:08:58.257 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=0, phi=1.0]
01-19 15:08:59.303 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=1, phi=1.4989764487358406]
01-19 15:09:00.176 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=2, phi=2.2460457009977275]
01-19 15:09:00.785 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=3, phi=3.3632275702400882]
01-19 15:09:01.344 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=4, phi=5.030355817286963]
01-19 15:09:01.770 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=5, phi=7.508649422981911]
01-19 15:09:02.210 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=6, phi=11.167302686223511]
01-19 15:09:02.567 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=7, phi=16.502140913586658]
01-19 15:09:02.892 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=8, phi=24.11844615397868]
01-19 15:09:03.172 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=9, phi=34.619498798804955]
01-19 15:09:03.422 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=10, phi=48.289086064804025]
01-19 15:09:03.743 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=11, phi=64.4074864453988]
01-19 15:09:03.973 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=12, phi=80.2535519764846]
01-19 15:09:04.178 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=13, phi=91.16357331407079]
01-19 15:09:04.386 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=14, phi=94.84633389963595]
01-19 15:09:04.586 127.0.0.1:40006       238       FJ-1-13  INFO water.default: Tweedie dispersion ML estimation [iter=15, phi=95.16374971030284]
01-19 15:09:04.776 127.0.0.1:40006       238       FJ-1-13  INFO water.default: last dispersion 95.16374971030284
```
**Local:**
```
01-19 17:27:35.834 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: GLM[dest=TweedieDispersionMLE_p1.6_phi100_offset0, iter=5 lmb=.0E0 alpha=.5E0 obj=105.36 imp=.11E-11 bdf=.49E-5] computed in 2+0=2ms, step = 1
01-19 17:27:35.835 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: estimating dispersion parameter using method: ml
01-19 17:27:38.027 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=0, phi=1.0]
01-19 17:27:38.425 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=1, phi=2.3134948275252825]
01-19 17:27:38.721 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=2, phi=3.476462736219342]
01-19 17:27:38.971 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=3, phi=5.214583849183841]
01-19 17:27:39.189 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=4, phi=7.8216170297537175]
01-19 17:27:39.364 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=5, phi=11.731807915358901]
01-19 17:27:39.524 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=6, phi=17.59620815067459]
01-19 17:27:39.648 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=7, phi=26.390576074000453]
01-19 17:27:39.756 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=8, phi=39.576413305657084]
01-19 17:27:39.855 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=9, phi=59.34041678495157]
01-19 17:27:39.948 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=10, phi=88.94842293324845]
01-19 17:27:40.051 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=11, phi=133.26394404988253]
01-19 17:27:40.139 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=12, phi=199.49823333185148]
01-19 17:27:40.220 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=13, phi=298.2754323688649]
01-19 17:27:40.291 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=14, phi=445.1002518415688]
01-19 17:27:40.368 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=15, phi=662.2707318919561]
01-19 17:27:40.435 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=16, phi=981.1034028905663]
01-19 17:27:40.506 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=17, phi=1443.8551290423538]
01-19 17:27:40.580 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=18, phi=2103.557843867063]
01-19 17:27:40.642 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=19, phi=3017.435895863103]
01-19 17:27:40.701 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion ML estimation [iter=20, phi=4224.820703774725]
01-19 17:27:40.749 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie sanity check FAIL. Trying Golden-section search instead of Newton's method.
01-19 17:27:40.995 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie golden-section search[iter=4, phis=(1.0E-16, 88.94842293324845, 2610.93919493278, 4224.820703774725), likelihoods=(..., -43173.961897211746, -55981.606199679714, ...)]
...
01-19 17:27:41.457 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie golden-section search[iter=45, phis=(101.69161031725578, 101.69161466635309, 101.69161735322994, 101.69162170232725), likelihoods=(..., -43131.49292150597, -43131.4929215061, ...)]
01-19 17:27:41.457 192.168.0.73:54321    4096       FJ-1-7  INFO water.default: Tweedie dispersion estimate = 101.69161383524286
```